kne_cand_vetting/survey_phot.py

```diff
@@ -384,8 +384,11 @@ def TNS_get(get_obj, BOT_ID: str = None, BOT_NAME: str = None, API_KEY: str = No
     # check the response to make sure it isn't throttling our API usage
     while True:
         response = requests.post(get_url, headers = headers, data = get_data)
+        if response.status_code != 200:
+            return response, -99 # I'm just setting the time_to_reset to -99 so other code doesn't break
+        
         remaining_str = response.headers.get('x-rate-limit-remaining')
-        time_to_reset = int(response.headers.get('x-rate-limit-reset')) # in seconds
+        time_to_reset = int(response.headers.get('x-rate-limit-reset', -99)) # in seconds
```

Comment:
Are you sure -99 is a good choice for this placeholder? I'm worried that we are going to try to sleep for -99 seconds in line 402. Maybe we actually want 0, i.e., try again immediately.

Reply:
I chose -99 so that it sticks out in the log/slack messages and we know what is going wrong. If we put in 0, I'm worried it will just look like "oh we ran out of API time" in the slack message. But, that being said, I see your point. When writing the code, I thought it would be fine because this elif statement will execute and we will see the -99 in the logs and get a hint at what happened. To be redundant, I just pushed a change that will check if the `time_to_wait` or `remaining_str` is -99 and exit the TNS checking then too. Although, I don't immediately see how the code would ever get to this point because it should only happen if 1) TNS changes their JSON format (unlikely) or 2) TNS is down and the request response status code is not 200.